### ISSUE TITLE
💬 RFC: Bumping `@kubernetes/client-node`to newest version (1.0.0)

### ISSUE THREAD

**Issue by elaine-mattos** [backstage/backstage#28325]
### 🔖 Need

Hi all,

I’m not entirely sure if this qualifies as a bug, so I’ve opened this as an RFC to start the discussion.

We recently attempted to bump the `@kubernetes/client-node` package to its newest version (1.0.0). However, we encountered the following error for every internal plugin that uses this client:
```css
Error [ERR_REQUIRE_ESM]: require() of ES Module...
```
Looking further, it seems that the `@kubernetes/client-node` switched to using ES Modules (ESM) starting from version 1.0.0, as noted [here](https://github.com/kubernetes-client/javascript/issues/2117#issuecomment-2555695892) and in its dependency `openid-client`([breaking changes](https://github.com/panva/openid-client/releases/tag/v6.0.0)).

This breaking change appears to cause compatibility issues with Backstage plugins that still rely on CommonJS imports. It seems this may require a refactor in how the package is imported within the Kubernetes plugin.

I was wondering if there is already any ongoing effort or plan to address this? If not, would love to hear thoughts on how best to approach this.

Thanks in advance!

### 🎉 Proposal

1. Refactoring the relevant Backstage plugins (e.g., Kubernetes plugin) to support ESM imports for `@kubernetes/client-node`.
2. Ensuring backward compatibility with existing functionality if possible, or providing clear migration guidance for users of these plugins.

### 👀 Have you spent some time to check if this RFC has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

**Comment by freben** [comment]
#12218 is where this is tracked. Note the PR at the bottom there, working toward this. Feel free to close this issue and focus the discussion in that issue if you agree that it covers your ask.

**Comment by elaine-mattos** [comment]
Hi @freben ,

thanks a LOT for the heads up!
I have managed to make it work locally using node 22, the `--experimental-require-module` flag and `@kubernetes/client-node`1.0.0. Also had to update `better-sqlite3` from 8.7.0 to 11.7.0 to support node 22. 
Couldn't make it work for node 23 as there were issues with `better-sqlite3` (I believe there isn't a build for node 23 yet, not really sure tho)...
I tested the Scaffolder by running some templates but I haven't performed any end to end tests yet.  I'll talk to the team at the beggining of the year and if we decide to go for this config I'll post it in the https://github.com/backstage/backstage/issues/12218 thread.

**Comment by elaine-mattos** [comment]
Hi @freben ,

thanks a LOT for the heads up!
I have managed to make it work locally using node 22, the `--experimental-require-module` flag and `@kubernetes/client-node`1.0.0. Also had to update `better-sqlite3` from 8.7.0 to 11.7.0 to support node 22. 
Couldn't make it work for node 23 as there were issues with `better-sqlite3` (I believe there isn't a build for node 23 yet, not really sure tho)...
I tested the Scaffolder by running some templates but I haven't performed any end to end tests yet.  I'll talk to the team at the beggining of the year and if we decide to go for this config I'll post it in the https://github.com/backstage/backstage/issues/12218 thread.
